src/lib/transaction_database/transaction_database.ml

```diff
@@ -43,15 +43,35 @@ struct
 
   type t = {database: Database.t; cache: cache; logger: Logger.t}
 
-  let create logger directory =
-    { database= Database.create ~directory
-    ; cache=
-        { user_transactions= Public_key.Compressed.Table.create ()
-        ; all_transactions= Transaction.Table.create ()
-        ; delegators= Public_key.Compressed.Table.create () }
-    ; logger }
+  let choose_delegator delegators ((txn, _) as txn_with_time) =
```

Comment:
this should be computed from the ledger directly rather than from the transaction db as the stake delegation could still be in progress or fail, etc.